### ISSUE TITLE
`unnecessary_parenthesis` false positive with `records`

### ISSUE THREAD

**Issue by asashour** [dart-lang/linter#4060]
```dart
const a = 1;
const b = 2;

void f(int i) {
  switch (i) {
    case const (a + b):
      break;
  }
}
```

There is a lint of `unnecessary_parenthesis`, however removing them causes a compilation error.

**Comment by srawlins** [comment]
Great catch. CC @pq

**Comment by pq** [comment]
Thank you!

**Comment by pq** [comment]
See also #4062
